bundle/metadata/dependencies.yaml

```diff
@@ -11,7 +11,33 @@ dependencies:
     value:
       packageName: dns-operator
       version: "0.0.0"
-  - type: olm.package
+  - type: olm.gvk
```

Comment:
According to 
https://github.com/Kuadrant/kuadrant-operator/blob/dc98cab7d131931351b32cdcf792d26442240ad0/controllers/tlspolicy_controller.go#L50-L53

```
//+kubebuilder:rbac:groups="cert-manager.io",resources=issuers,verbs=get;list;watch;
//+kubebuilder:rbac:groups="cert-manager.io",resources=clusterissuers,verbs=get;list;watch;
//+kubebuilder:rbac:groups="cert-manager.io",resources=certificates,verbs=get;list;watch;create;update;patch;delete
```

`CertificateRequest`, `Order` and `Challenge` GVKs are not being used by the TLSPolicy controller. Should those also be added as deps? @maleck13 @mikenairn

Reply:
I wouldn't have thought we needed to add GVKs for things this controller doesn't deal with directly, but then again I'm not totally sure what OLM expects you to do here, if anything. Why do we want to change it from specifying a known package to specifying gvks that any package could be providing?